### PR TITLE
New developers docs (Part 1: Index)

### PR DESCRIPTION
**Pull request by mliradelc**
The new index is ready for review.
This will be the new developer's documentation for versions 14 and onwards (Also apply for previous versions)

Tasks:
* [x] Write new Index

### CODE REVIEW

File: docs/guides/developer/mkdocs.yml [hunk 1]
```diff
@@ -27,79 +27,89 @@ extra_javascript:
...
+         - Index: 'develop/admin-ui/style/index.md'
+         - Modals: 'develop/admin-ui/style/modals.md'
+         - Navigation: 'develop/admin-ui/style/navigation.md'
+         - References: 'develop/admin-ui/style/references.md'
+         - Spacing: 'develop/admin-ui/style/spacing.md'
+         - Typeface: 'develop/admin-ui/style/typeface.md'
+         - View Structure: 'develop/admin-ui/style/view-structure.md'
+   - Documentation: 'develop/documentation.md'
+   - Localization: 'develop/localization.md'
+- Opencast architecture:
```
Comment (JamesUoM) [inline docs/guides/developer/mkdocs.yml:84]: We should have documentation on the workspace or even more generally on the filesystem layout

Comment (mliradelc) [reply docs/guides/developer/mkdocs.yml:84]: Done! I added the `filesystem` page. 

Thanks for the suggestion!

File: docs/guides/developer/docs/develop/setup-opencast-for-develop.md
Comment (gregorydlogan) [inline docs/guides/developer/docs/develop/setup-opencast-for-develop.md:1]: IMO this file needs to be heavily revised, formatting wise.  For a later PR though.

File: docs/guides/developer/mkdocs.bkp.yml
Comment (gregorydlogan) [inline docs/guides/developer/mkdocs.bkp.yml:1]: Extraneous file, should be removed.

**Review by gregorydlogan — CHANGES_REQUESTED** [review]
Looks good to me, very minor nitpick wrt to the extra backup(?) file that got included accidentally.

File: docs/guides/developer/mkdocs.bkp.yml
Comment (mliradelc) [reply docs/guides/developer/mkdocs.bkp.yml:1]: Yes, you have right, I will delete it. :)

**Pull request merged**